### PR TITLE
Patch BlockDynamicLiquid#isBlocked to use state sensitive material check

### LINKED ISSUE #4473: BlockDynamicLiquid Using State Insensitive Material Check
`BlockDynamicLiquid#isBlocked` does not use the state sensitive `IBlockProperties#getMaterial` for its material checks. This causes issues for blocks that change their material based on their state values. I posted here as I feel it could be easily patched; although I don't know how to write patches.

Edit: I've been informed on how patching works and I'll try to make a PR.

**Minecraft version:** 1.12.2
**Forge version:** 14.23.0.2512

### PR DESCRIPTION
**Pull request by ChloeDawn**
This pull request resolves #4473

### CODE REVIEW

**Comment by ChloeDawn** [comment]
Should the state be declared once in its own field for this? I'm not sure how much it would affect performance declaring it twice for `material` and `block` individually.

**Comment by covers1624** [comment]
Is this really needed no-one should have custom impl's of BlockDynamicLiquid as forge has it's own fluid block base class.

**Comment by ChloeDawn** [comment]
It causes issues when a mod provides different block materials based on state values. Vanilla's fluid block is currently only respecting the material passed in the super of the Block constructor, which I personally am not relying on for some of my blocks, that use a PropertyEnum for variants, all of which pass their own material to the state sensitive method in Block.

**Comment by ChloeDawn** [comment]
For example passing `Material.STONE` to the constructor and then having a state variant that passes `Material.CIRCUITS` to `Block#getMaterial(IBlockState)`, will be treated as `Material.STONE` by the `BlockDynamicLiquid#isBlocked`

**Comment by covers1624** [comment]
Right, but why use BlockDynamicLiquid over forges implementation?

**Comment by ChloeDawn** [comment]
I am not. But vanilla uses it for water and lava. This method not being state sensitive with its material check causes problems between modded blocks and vanilla fluids.

**Comment by covers1624** [comment]
Ohhhh, was confused as to what the method actually did, don't have code infront of me.. in that case patch looks good to me.

File: patches/minecraft/net/minecraft/block/BlockDynamicLiquid.java.patch [hunk 1]
```diff
@@ -17,3 +17,17 @@
                      p_176375_3_.func_177230_c().func_176226_b(p_176375_1_, p_176375_2_, p_176375_3_, 0);
                  }
              }
+@@ -254,11 +255,12 @@
+ 
+     private boolean func_176372_g(World p_176372_1_, BlockPos p_176372_2_, IBlockState p_176372_3_)
+     {
++        Material material = p_176372_1_.func_180495_p(p_176372_2_).func_185904_a();
```
Comment (mezz) [inline patches/minecraft/net/minecraft/block/BlockDynamicLiquid.java.patch]: Please move this line down so the two patched lines are right next to each other

**Review by mezz — CHANGES_REQUESTED** [review]
Makes sense, just needs a minor change to the patch to make it smaller and simpler to update.

**Review by mezz — APPROVED** [review]

**Pull request closed without merging**